File: 202.Happy_Number.java
class Solution {
    public boolean isHappy(int n) {
        HashSet<Integer> exist = new HashSet();
        while (exist.add(n)) {
            int squareSum = 0;
            while (n != 0){
                squareSum += Math.pow(n%10, 2);
                n /= 10;
            }
            if (squareSum == 1)
                return true;
            n = squareSum;
        }
        return false;
    }
}

File: 202.HappyNumber.py
class Solution:
    def isHappy(self, n: int) -> bool:
        hs = set()
        while n > 1:
            sum = 0
            while n > 0:
                digit = n%10
                n //= 10
                sum += digit*digit
            if sum in hs:
                return False
            hs.add(sum)
            n = sum
        return True
        